redis/connection.js

```diff
@@ -6,87 +6,93 @@
 
 const Promise = require('../promise');
 
-module.exports = (log, client) => {
-  let isUpdating = false;
-  let destroyPromise;
-
-  return {
-    get (key) {
-      return client.getAsync(key);
-    },
-
-    set (key, value) {
-      return client.setAsync(key, value);
-    },
-
-    del (key) {
-      return client.delAsync(key);
-    },
-
-    /**
-     * To ensure safe update semantics in the presence of concurrency,
-     * we lean on Redis' WATCH, MULTI and EXEC commands so that updates
-     * run as a transaction and will fail if the data changes underneath
-     * them. You can read more about this in the Redis docs:
-     *
-     *   https://redis.io/topics/transactions
-     *
-     * @param key {String} The key to update
-     * @param getUpdatedValue {Function} A callback that receives the current value
-     *                                   and returns the updated value. May return a
-     *                                   promise or the raw updated value.
-     */
-    async update (key, getUpdatedValue) {
-      if (isUpdating) {
-        log.error('redis.update.conflict', { key });
-        throw new Error('redis.update.conflict');
-      }
+// Redis command reference: https://redis.io/commands
+const SUPPORTED_COMMANDS = [
+  // Basic operations
+  'get', 'set', 'del',
+  // Sorted sets: https://redis.io/topics/data-types-intro#redis-sorted-sets
+  'zadd', 'zcard', 'zcount', 'zpopmax', 'zpopmin', 'zrange', 'zrangebyscore', 'zrank', 'zrem',
+  'zremrangebyrank', 'zremrangebyscore', 'zrevrange', 'zrevrangebyscore', 'zrevrank', 'zscore',
+];
+
+module.exports = {
+  methods: SUPPORTED_COMMANDS.concat('update'),
+
+  create (log, client) {
+    let isUpdating = false;
+    let destroyPromise;
+
+    return {
+      ...SUPPORTED_COMMANDS.reduce((object, command) => {
+        object[command] = (...args) => client[`${command}Async`](...args);
```

Comment:
All of those have `Async` equivalents? Should we bail early if an expected method does not exist?

Reply:
They all exist.